src/login/logind-core.c

```diff
@@ -646,6 +648,13 @@ bool manager_is_docked_or_external_displays(Manager *m) {
                 log_warning_errno(n, "Display counting failed, assuming system is not docked: %m");
         else if (n >= 1) {
                 log_debug("System is docked (%i external display%s connected).", n, n > 1 ? "s" : "");
+                m->external_monitor_timestamp = now(CLOCK_MONOTONIC);
```

Comment:
this doesn't look right: the displays might have been plugged in a long time ago, and we'll wait here needlessly for a while for those. 

the correct way would probably be to install some sd-device match for these devices, and watch uevents to see when this changes (under the assumption the DRM devices actually synthesize uevents for changes like this)